recipe/meta.yaml

```
@@ -10,7 +10,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 1
+  number: 0
   run_exports:
     - {{ pin_subpackage("azure-storage-common-cpp", max_pin="x") }}
```

Actually, looking the failure on the other packages' feedstocks for PRs updating version. It seems that azure-storage-common API is changing on minor version (see [this error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=779589&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=220)).

Related PRs:
 - https://github.com/conda-forge/azure-storage-queues-cpp-feedstock/pull/2
 - https://github.com/conda-forge/azure-storage-files-datalake-cpp-feedstock/pull/4
 - https://github.com/conda-forge/azure-storage-blobs-cpp-feedstock/pull/4
 - https://github.com/conda-forge/azure-storage-queues-cpp-feedstock/pull/2
 
Hence I would change the runtime requirements to:

```suggestion
    - {{ pin_subpackage("azure-storage-common-cpp", max_pin="x.x") }}
```

---

Thanks for investigating. That is frustrating. However, I did see that they at least bumped the minor version requirement in `vcpkg.json` to `>= 12.4.0`, eg https://github.com/Azure/azure-sdk-for-cpp/blob/azure-storage-blobs_12.9.0/sdk/storage/azure-storage-blobs/vcpkg/vcpkg.json#L17. So we definitely need to build azure-storage-common-cpp 12.4.0 first and then bump the minimum required version in the requirements of the downstream recipes.

I'm open to also bumping the `run_exports` to `x.x`, but to make sure we're on the same page, it wouldn't have prevented these build failures, since the downstream packages require 12.4.0 to build no matter the value of `run_exports`

---

> the downstream packages require 12.4.0 to build no matter the value of run_exports

This is indeed a necessary condition, but for the future we need to have runtime requirements on minor versions, I believe.

@Shelnutt2: what do you think?

---

> that `azure-storage-common-cpp` API is changing on minor version

Does `max_pin="x"` mean that the API cannot break across minor versions or that it cannot change at all on minor versions? Unless it is the latter, I don't see a reason to diverge from the version compatibility the Azure SDK claims to offer.

---

`max_pin="x"` sets a runtime requirement to use this version or the next ones sharing the same major number at runtime — see [this section of the documentation](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#export-runtime-requirements)

It is typically used for API stability but the API can be extended in the meantime.

The errors might be evidence that the requirement is not strict enough: if this is the case we must constrain on the minor version number with `max_pin="x.x"`.

---

In the case of API addition, stricter runtime requirements on minor versions should not be needed since runtime requirements block the usage of previous versions.

I am currently trying to think of cases where `max_pin="x.x"` would be needed.

---

I think we can merge without the suggestion.